h/browser/chrome/lib/uri-info.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * A wrapper for the server's uriinfo endpoint.
+ *
+ * Exports a function that makes an HTTP request to the server's uriinfo
+ * endpoint and returns a Promise that resolves to the response. Usage:
+ *
+ *     var uriInfo = require('uri-info');
+ *     uriInfo(uri).then(function(info) {
+ *       ...
+ *     });
+ *
+ * This function can be called with the same uri multiple times consecutively
+ * and it'll just keep returning the same Promise - won't make multiple HTTP
+ * requests.
+ *
+ * But if you call the function with a different uri, then call it with the
+ * first uri again, then it _will_ make another request for the first uri.
+ *
+ * Returns a rejected Promise if the HTTP request times out, fails or is
+ * invalid for any reason.
+ *
+ * Returns a rejected Promise is the given uri is undefined.
+ *
+ * @param {String} uri The URI to be checked
+ *
+ * @returns {Promise} A Promise that resolves to an object with two properties:
+ *   1. total:   the number of annotations there are of this URI (Number)
+ *   2. blocked: whether or not this URI is blocklisted (Boolean)
+ *
+ *
+ * @module
+ */
+'use strict';
+
+var settings = require('./settings');
+var uriInfoPromise;
+
+/**
+  * Make an HTTP request to the uriinfo endpoint and return a Promise that
+  * resolves to the value of the response.
+  *
+  */
+function getUriInfo(uri) {
+  if (uri === undefined) {
```

Comment:
For consistency, I'd like to use falsey-ness everywhere rather than explicitly checking for a particular falsey value, except where the difference between an empty string / null and undefined matters.